### PR TITLE
Login with trakt and sync history in movies

### PR DESCRIPTION
**Pull request by hadi-norouzi**
I've implemented these features:

1. login with Trakt in web-view and get access token
2. get movie history by TraktID
3. Now user can add movies to Trakt history

if you have any feedback, I would be happy to hear it.

#16

### CODE REVIEW

**Comment by moallemi** [comment]
Dear @hadi-norouzi, just wanted to say a big thanks for the awesome pull request you dropped in this repo. It's clear you've put a lot of effort into it, and it really stands out. Can't wait to dig into the details!

Currently, I have some urgent commitments to attend to this week, which means I won't be able to get to your PR right away. However, I want to assure you that your PR is a priority for me, and I plan to allocate dedicated time for a review as soon as possible. I expect to be able to do this by the end of next week.

Thanks again for your hard work and dedication to the project. Your contributions are really valuable to me.

**Comment by moallemi** [comment]
Dear @hadi-norouzi, just wanted to say a big thanks for the awesome pull request you dropped in this repo. It's clear you've put a lot of effort into it, and it really stands out. Can't wait to dig into the details!

Currently, I have some urgent commitments to attend to this week, which means I won't be able to get to your PR right away. However, I want to assure you that your PR is a priority for me, and I plan to allocate dedicated time for a review as soon as possible. I expect to be able to do this by the end of next week.

Thanks again for your hard work and dedication to the project. Your contributions are really valuable to me.

**Comment by hadi-norouzi** [comment]
> Dear @hadi-norouzi, just wanted to say a big thanks for the awesome pull request you dropped in this repo. It's clear you've put a lot of effort into it, and it really stands out. Can't wait to dig into the details!
> 
> Currently, I have some urgent commitments to attend to this week, which means I won't be able to get to your PR right away. However, I want to assure you that your PR is a priority for me, and I plan to allocate dedicated time for a review as soon as possible. I expect to be able to do this by the end of next week.
> 
> Thanks again for your hard work and dedication to the project. Your contributions are really valuable to me.

It's my pleasure to work with you and I wish to be more valuable to this project.

**Review by moallemi — APPROVED** [review]
Dear @hadi-norouzi 

I just wanted to say I'm really sorry for getting to your PR so late. I know you put effort into this, and it wasn't cool to keep you waiting. Thanks for your patience.

Currently, you might not be in the mood to revise and update the existing PR, so I kindly request that you simply create issues where I've noted in the comments. This way, we can enhance the code later based on these new issues.

Best,
Reza

File: data/api/trakt/src/main/java/io/filmtime/data/api/trakt/TraktAuthRemoteSourceImpl.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+class TraktAuthRemoteSourceImpl @Inject constructor(
+  private val traktAuthService: TraktAuthService,
+) : TraktAuthRemoteSource {
+
+  override suspend fun getAccessToken(code: String): Result<TraktTokens, GeneralError> {
+    val result = traktAuthService.getAccessToken(
+      body = TraktGetTokenRequest(
+        code = code,
+        clientID = BuildConfig.TRAKT_CLIENT_ID,
+        clientSecret = BuildConfig.TRAKT_CLIENT_SECRET,
```
[the remark this answers is not in the document]
Comment (hadi-norouzi) [reply data/api/trakt/src/main/java/io/filmtime/data/api/trakt/TraktAuthRemoteSourceImpl.kt:21]: I moved other apis to set headers on interceptor but this function sends these fields as body request.

File: data/api/trakt/src/main/java/io/filmtime/data/api/trakt/TraktSyncRemoteSourceImpl.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  override suspend fun getHistoryById(id: String): Result<Boolean, GeneralError> {
+    // TODO: move check token in a function
+    val tokens =
+      traktAuthLocalSource.tokens.firstOrNull() ?: return Result.Failure(GeneralError.ApiError("Unauthorized", 401))
+    val result = traktSyncService.getHistoryById(
+      type = "movies",
+      id = id,
+      accessToken = "Bearer " + tokens.accessToken,
```
[the remark this answers is not in the document]
Comment (hadi-norouzi) [reply data/api/trakt/src/main/java/io/filmtime/data/api/trakt/TraktSyncRemoteSourceImpl.kt:41]: Is it okay to depend network module on LocalStorage module to read stored tokes?

**Comment by hadi-norouzi** [comment]
> Dear @hadi-norouzi
> 
> I just wanted to say I'm really sorry for getting to your PR so late. I know you put effort into this, and it wasn't cool to keep you waiting. Thanks for your patience.
> 
> Currently, you might not be in the mood to revise and update the existing PR, so I kindly request that you simply create issues where I've noted in the comments. This way, we can enhance the code later based on these new issues.
> 
> Best, Reza
I've no problem at all. I've just waiting for your youtube videos 😄.

**Review by moallemi — APPROVED** [review]

**Comment by moallemi** [comment]
@hadi-norouzi, could you kindly reach out to me via Telegram (@moallemi) or Discord to discuss a few matters?

**Pull request closed without merging**